src/nvim/memline.c

```diff
@@ -2492,6 +2492,11 @@ int ml_replace(linenr_T lnum, char *line, bool copy)
   return ml_replace_buf(curbuf, lnum, line, copy, false);
 }
 
+int ml_replace_len(linenr_T lnum, char *line, size_t len, bool copy)
```

Comment:
```
src/proto/memline.pro:33:int ml_replace_len(linenr_T lnum, char_u *line_arg, colnr_T len_arg, int has_props, int copy);
```

`has_props` being bool complicates porting since it's N/A for Nvim.